# IMPORTING MODULES

In [1]:
import json as js

# Openning The Old Inventory

In [2]:
f1 = open("record.json","r")
txt= f1.read()
f1.close()
record = js.loads(txt)
for k,v in record.items():
    print("Product UID: {} => Product Details {}".format(k,v))
print("\nTotal Number of product: ",len(record))

Product UID: 1120770 => Product Details {'name': 'Kurkure', 'type': 'chips', 'price': 20, 'quantity': 500, 'weightinGM': 40}
Product UID: 1120771 => Product Details {'name': 'Perk', 'type': 'chocolate', 'price': 5, 'quantity': 1000, 'weightinGM': 200}
Product UID: 1120772 => Product Details {'name': 'Tatatea', 'type': 'tea', 'price': 50, 'quantity': 400, 'weightinGM': 500}
Product UID: 1120773 => Product Details {'name': 'RedLabel', 'type': 'tea', 'price': 90, 'quantity': 932, 'weightinGM': 240}
Product UID: 1120774 => Product Details {'name': 'Bourbon', 'type': 'Biscuit', 'price': 30, 'quantity': 450, 'weightinGM': 10}
Product UID: 1120775 => Product Details {'name': 'ParleG', 'type': 'Biscuit', 'price': 10, 'quantity': 300, 'weightinGM': 100}
Product UID: 1120776 => Product Details {'name': 'Oreo', 'type': 'Biscuit', 'price': 50, 'quantity': 230, 'weightinGM': 150}
Product UID: 1120777 => Product Details {'name': 'Hide&Seek', 'type': 'Biscuit', 'price': 30, 'quantity': 534, 'weightin

# Adding new product in the record or Updtaing the old inventory

In [7]:
product_id = input('Enter product ID(New or Old): ')
product_name = input('Enter product name(New or Old): ')
types = input('Enter product type: ')
quantity = int(input('Enter product quantity(*replaces previous quantity*): '))
price = int(input('Enter product price: '))
weightinGM = int(input('Enter product weigth: '))
y = {product_id:{'name':product_name,'type':types,'price':price,'quantity':quantity,'weightinGM':weightinGM}}
record.update(y)

Enter product ID(New or Old): 1120779
Enter product name(New or Old): DairyMilkSilk
Enter product type: chocolate
Enter product quantity(*replaces previous quantity*): 400
Enter product price: 120
Enter product weigth: 210


# Updating the inventory file

In [8]:
jsup = js.dumps(record)
f2 = open("record.json","w")
xw = f2.write(jsup)
print("Report Updated successfully")
f2.close()

Report Updated successfully


# Checking the updated inventory

In [10]:
f = open("record.json","r")
txt= f.read()
f.close()
record = js.loads(txt)
for k,v in record.items():
    print("Product UID: {} => Product Details {}".format(k,v))

Product UID: 1120770 => Product Details {'name': 'Kurkure', 'type': 'chips', 'price': 20, 'quantity': 500, 'weightinGM': 40}
Product UID: 1120771 => Product Details {'name': 'Perk', 'type': 'chocolate', 'price': 5, 'quantity': 1000, 'weightinGM': 200}
Product UID: 1120772 => Product Details {'name': 'Tatatea', 'type': 'tea', 'price': 50, 'quantity': 400, 'weightinGM': 500}
Product UID: 1120773 => Product Details {'name': 'RedLabel', 'type': 'tea', 'price': 90, 'quantity': 932, 'weightinGM': 240}
Product UID: 1120774 => Product Details {'name': 'Bourbon', 'type': 'Biscuit', 'price': 30, 'quantity': 450, 'weightinGM': 10}
Product UID: 1120775 => Product Details {'name': 'ParleG', 'type': 'Biscuit', 'price': 10, 'quantity': 300, 'weightinGM': 100}
Product UID: 1120776 => Product Details {'name': 'Oreo', 'type': 'Biscuit', 'price': 50, 'quantity': 230, 'weightinGM': 150}
Product UID: 1120777 => Product Details {'name': 'Hide&Seek', 'type': 'Biscuit', 'price': 30, 'quantity': 534, 'weightin

# CHECKING THE SIZE OF DICTIONARY(CART)

In [11]:
len(record)

32

#Reference of Dictionary

record = {1120770:{"name":"Kurkure", "type":"chips",     "price":20, "quantity":500,  "weightinGM":40},
          1120771:{"name":"Perk",    "type":"chocolate", "price":5,  "quantity":1000, "weightinGM":200},
          1120772:{"name":"Tatatea", "type":"tea",       "price":50, "quantity":400,  "weightinGM":500},
          1120773:{"name":"RedLabel","type":"tea",       "price":90, "quantity":940,  "weightinGM":240},
          1120774:{"name":"Bourbon", "type":"Biscuit",   "price":30, "quantity":450,  "weightinGM":10},
          1120775:{"name":"ParleG",  "type":"Biscuit",   "price":10, "quantity":300,  "weightinGM":100},
          1120776:{"name":"Oreo",    "type":"Biscuit",   "price":50, "quantity":230,  "weightinGM":150},
          1120777:{"name":"Hide&Seek","type":"Biscuit",  "price":30, "quantity":540,  "weightinGM":100},
          1120778:{"name":"HappyHappy","type":"Biscuit", "price":20, "quantity":140,  "weightinGM":50},
          1120779:{"name":"DairyMilkSilk","type":"chocolate","price":100,"quantity":100,"weightinGM":500},
          1120780:{"name":"KitKat",  "type":"chocolate", "price":40, "quantity":80,   "weightinGM":300},
          1120781:{"name":"5Star",   "type":"chocolate", "price":20, "quantity":120,  "weightinGM":140},
          1120782:{"name":"Gems",    "type":"chocolate", "price":10, "quantity":200,  "weightinGM":80},
          1120783:{"name":"balajiOnion","type":"chips","price":30,"quantity":480,     "weightinGM":240},
          1120784:{"name":"laysSpices","type":"chips","price":40,"quantity":340,      "weightinGM":140},
          1120785:{"name":"CocaCola","type":"coldDrinks","price":50,"quantity":140,   "weightinGM":1000},
          1120786:{"name":"Maza", "type":"coldDrinks","price":20,"quantity":340,      "weightinGM":500},
          1120787:{"name":"Sprit","type":"coldDrinks","price":30,"quantity":240,      "weightinGM":250},
          1120788:{"name":"pepssi",  "type":"coldDrinks","price":50,"quantity":440,   "weightinGM":200},
          1120789:{"name":"Colgate", "type":"paste","price":50,"quantity":640,        "weightinGM":140},
          1120790:{"name":"Pepsudent","type":"paste","price":30,"quantity":220,       "weightinGM":140},
          1120791:{"name":"Wheel",   "type":"soap", "price":5, "quantity":1200,       "weightinGM":100},
          1120792:{"name":"Vim",     "type":"soap", "price":10, "quantity":600,       "weightinGM":250},
          1120793:{"name":"Dettol",  "type":"soap","price":20,"quantity":480,         "weightinGM":200},
          1120794:{"name":"Fogg",    "type":"perfume","price":120,"quantity":2340,    "weightinGM":250},
          1120795:{"name":"Denver",  "type":"perfume","price":100,"quantity":340,     "weightinGM":250},
          1120796:{"name":"spark",   "type":"perfume","price":100,"quantity":740,     "weightinGM":250},
          1120797:{"name":"Stylo",   "type":"pen","price":10,"quantity":1240,         "weightinGM":30},
          1120798:{"name":"Nattaraj","type":"pen","price":10,"quantity":440,          "weightinGM":20},
          1120799:{"name":"Apasra",  "type":"pencil","price":7,"quantity":640,        "weightinGM":30},
          11207100:{"name":"HBO",    "type":"pencil","price":10,"quantity":120,       "weightinGM":15},
          11207101:{"name":"Doom",    "type":"pencil","price":10,"quantity":200,       "weightinGM":10}
          
}

# Thank You